roles/hosted_engine_setup/tasks/create_target_vm/03_hosted_engine_final_tasks.yml

```diff
@@ -184,7 +184,7 @@
     args:
       warn: false
   - name: Initialize metadata volume
-    command: dd bs=1M count=1024 oflag=direct if=/dev/zero of="{{ he_metadata_disk_path }}"
+    command: fallocate -l 1G "{{ he_metadata_disk_path }}"
```

Comment:
Is this a disk on a storage domain, so it can be file on NFS 3, NFS 4.2, Gluster
or a block device?

Here is fallocate support matrix:
- Glsuter: works
- NFS: works only with NFS 4.2. "fallocate --posix" works with any NFS version, but is extremely slow and problematic, vdsm never use it.
- Block:
  - "fallocatge --posix" does not work, but also does not fail (looks like a bug)
  - "fallocate --punch-hole" should work, but it can fail with EOPNOTSUPP
  - "fallocate --zero-range" should work, but it can fail with EOPNOTSUPP
  - "blkdiscard --zeroout" work with any block device (what vdsm uses)

Vdsm uses:
- File: /usr/libexec/vdsm/fallocate - helper using fallocate(2) or fall back to manual zeroing
- Block: "blkdiscard --zeroout --step 32m"

I think we need to understand first why do you need 1g. Most likely you need only few blocks and using dd is the right way to zero them.

Reply:
@tiraboschi do you have any idea why we initialize 1g?

Reply:
Data is written at offset 4KiB*host_id, so we don't really need to initialize 1GiB, changed to 1MiB.